src/pip/_internal/commands/install.py

```diff
@@ -446,7 +452,7 @@ def run(self, options: Values, args: List[str]) -> int:
                 wheel_cache=wheel_cache,
                 verify=True,
                 build_options=[],
-                global_options=[],
+                global_options=global_options,
```

Comment:
Hmm this one’s unexpected. Why was it `[]`…?

Reply:
Actually, the only way to reach this point when `--global-option` is present, is to `--use-pep517`.
By passing it here, we trigger the [warning](https://github.com/pypa/pip/blob/0a21080411c25acfb87fbc380631806e0477d7d3/src/pip/_internal/wheel_builder.py#L245-L248) in wheel_builder that global options are ignored when doing pep 517 builds.
This should help nudging users to use `--config-settings`.

Reply:
I guess what I’m wondering is mostly, was it a bug `global_options` wasn’t passed previously?

Reply:
> was it a bug

Yes and no. The only "bug" was that --global-option was ignored *without warning* in pep 517 mode.